src/metabase/lib/cache.cljc

```diff
@@ -58,46 +97,49 @@
     being a vanilla JS object or a CLJS map. (Eg. if the `host` is an instance of a JS class.)"
   [subkey host x f {:keys [cache-fn force?]
                     :or {cache-fn atomic-map-cache-fn}}]
-  (comment subkey, force?, cache-fn, host) ; Avoids lint warning for half-unused inputs.
-  #?(:clj  (f x)
-     :cljs (if (or force? (object? host) (map? host))
-             (do
-               (when-not (.-__mbcache ^js host)
-                 (set! (.-__mbcache ^js host) (cache-fn)))
-               (if-let [cache (.-__mbcache ^js host)]
-                 (if-let [cached (cache-fn cache subkey x)]
-                   cached
-                   ;; Cache miss - generate the value and cache it.
-                   (let [value (f x)]
-                     (cache-fn cache subkey x value)
-                     value))
-                 (f x)))
-             (f x))))
+  (if (or force?
+          (map? host)
+          #?(:cljs (object? host)
+             :clj  false))
```

Comment:
Why is the `:clj` branch needed?

Reply:
I suppose it's not; `nil` would have worked just as well.

Reply:
I expect it to be completely omitted in CLJ if there is no `:clj` clause, so not even a `nil`.

Reply:
That's not how `#?` works; it resolves to `nil` in dialects which are not included in the conditional. That's fine here, though.

That's why we use `#?@(:cljs ((:require-macros ...)))` with a one-element list - `nil` isn't allowed in `ns` macros and splicing in `nil` is an empty seq, so it disappears.

Reply:
Why do I see
```
user> (count [0 #?(:clj 1) #?(:cljs 2)])
2
```
in my REPLs? Shouldn't it be `3`?

Reply:
Huh, I was very confident in that. Maybe there's some context where they work differently, but it certainly does seem like `#?(:cljs 7)` vanishes in CLJ and vice versa. TIL!